### PR TITLE
feat: add sse payload attributes example and beacon api support

### PR DESCRIPTION
**Pull request by mattsse**
adds an example of how to subscribe to the SSE endpoint

adds some additional serde support for the beacon API.

the format differs...

### CODE REVIEW

**Review by rkrasiuk — APPROVED** [review]
lgtm

File: crates/rpc/rpc-types/src/eth/withdrawal.rs [hunk 1]
```diff
@@ -25,6 +29,72 @@ impl Withdrawal {
     }
 }
 
+/// Same as [Withdrawal] but respects the Beacon API format which uses snake-case and
```
Comment (Rjected) [inline crates/rpc/rpc-types/src/eth/withdrawal.rs:32]: comment should be completed

File: crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+/// Event for the `payload_attributes` topic of the beacon API node event stream.
+///
+/// This event gives block builders and relays sufficient information to construct or verify a block
+/// at `proposal_slot`.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayloadAttributesEvent {
+    /// the identifier of the beacon hard fork at `proposal_slot`, e.g `"bellatrix"`, `"capella"`.
+    pub version: String,
```
Comment (Rjected) [inline crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs:18]: an enum would make sense here right? could be done in a followup

**Review by Rjected — APPROVED** [review]
small nits, otherwise lgtm + good tests

**Pull request merged**